src/xutils.cpp

```diff
@@ -83,20 +84,106 @@ namespace xpyt
 namespace nlohmann
 {
 
-    py::object adl_serializer<py::object>::from_json(const json& j)
+    namespace detail
     {
-        py::module py_json = py::module::import("json");
+        py::object from_json_impl(const json& j)
+        {
+            if (j.is_null())
+            {
+                return py::none();
+            }
+            if (j.is_boolean())
+            {
+                return py::bool_(j.get<bool>());
+            }
+            if (j.is_number())
+            {
+                double number = j.get<double>();
+                if (number == std::floor(number))
+                {
+                    return py::int_(j.get<int>());
```

Comment:
Could use a note that this is `int` everywhere rather than `size_t` because of JSON representation limitation (IIUC).

Reply:
JSON and JS can only represent floating points, no real integer. The choice to use an integer when the value is integral is to be consistent with the behavior of the JSON module of Python.

Reply:
That's the behavior of the JSON module of Python. If the number value in the JSON can be represented as an integer value it creates an integer, not a floating point number. But the question is, is the `py::int_` an `int`, `long`, `long long` or `ptrdiff_t`?

Reply:
I guess it depends on the value itself... https://github.com/pybind/pybind11/blob/master/include/pybind11/pytypes.h#L1076